lib/storage.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const data = {};
+const Promise = require('bluebird');
+const storage = module.exports = {};
+const dataDir = `${__dirname}/../data`;
+const fs = Promise.promisifyAll(require('fs'));
+
+storage.setItem = function(name, item) {
+  return fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
+    err.status = 400;
+    return Promise.reject(err);
+  })
+  .then(() => {
+    let json = JSON.stringify(item);
+    return fs.writeFileAsync(`${dataDir}/${name}/${item.id}.json`, json);
+  })
+  .then(() => {
+    return item;
+  });
+};
+
+storage.getItem = function(name, id) {
+  return fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
```

Comment:
Underneath this catch block, you need to set err.status to 404 so you can get a proper 404 error if a user enters an invalid id number for a GET request. That 404 error will get passed on to the router. Right now, I'm only getting a 500 error, which isn't what we want.